src/OpenTelemetry.Exporter.Jaeger/JaegerExporterOptions.cs

```diff
@@ -14,14 +14,41 @@
 // limitations under the License.
 // </copyright>
 
+using System;
 using System.Diagnostics;
+using OpenTelemetry.Exporter.Jaeger.Implementation;
 
 namespace OpenTelemetry.Exporter
 {
     public class JaegerExporterOptions
     {
         internal const int DefaultMaxPayloadSizeInBytes = 4096;
 
+        private const string OTelAgentHostEnvVarKey = "OTEL_EXPORTER_JAEGER_AGENT_HOST";
+        private const string OTelAgentPortEnvVarKey = "OTEL_EXPORTER_JAEGER_AGENT_PORT";
+
+        public JaegerExporterOptions()
+        {
+            string agentHostEnvVar = Environment.GetEnvironmentVariable(OTelAgentHostEnvVarKey);
```

Comment:
I would move this section to a new method - eg `SetOptionsViaEnvironmentVariables`.

Reply:
Waiting for feedback (e.g. upvotes) from others as this may be subjective. This would be the only call in this constructor so it might be seen as an unnecessary level of indirection.

I have no strong opinion here.

Initially, I thought of doing something like:

```csharp
this.AgentHost = GetStringFromEnvironmentVariable(OTelAgentHostEnvVarKey, "localhost");
this.AgentPort = GetIntFromEnvironmentVariable(OTelAgentPortEnvVarKey, 6831, value => JaegerExporterEventSource.Log.FailedToParseEnvironmentVariable(OTelAgentPortEnvVarKey, value));
```

but it felt unnecessary complex

Reply:
I kind of like these "general" methods but deferring until more usage seems reasonable.

Reply:
Please wrap this inside try..catch incase user doesn't have permission to read, it'll throw.

Reply:
Will do

Reply:
Handled: 5909df24
@cijothomas Do we want to have another event for a generic `Exception`? Personally, I do not like it b/c we can catch e.g. an `OutOfMemoryException`.

Reply:
this looks good.